### PR TITLE
yii2 to yii3.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Tests pass?   | ✔️

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -55,5 +55,5 @@
     },
     "config": {
         "sort-packages": true
-    }
-}
\ No newline at end of file
+    },
```
Comment (samdark) [inline composer.json]: Should be removed.

File: composer.json [hunk 2]
```diff
@@ -18,20 +18,20 @@
...
         "yiisoft/arrays": "^3.0@dev",
-        "php": "^7.4"
+        "yiisoft/db": "^3.0@dev"
     },
     "require-dev": {
+        "hiqdev/composer-config-plugin": "^1.0@dev",
         "phpunit/phpunit": "^8.4",
-        "yiisoft/view": "^3.0@dev",
-        "yiisoft/cache": "^3.0@dev",
         "yiisoft/active-record": "^3.0@dev",
```
Comment (samdark) [inline composer.json]: Should not be used in this package tests.

File: composer.json [hunk 3]
```diff
@@ -18,20 +18,20 @@
...
+        "yiisoft/aliases": "^3.0@dev",
```
Comment (samdark) [inline composer.json]: Doesn't seem to be used.

File: composer.json [hunk 4]
```diff
@@ -18,20 +18,20 @@
...
+        "yiisoft/cache": "^3.0@dev",
```
Comment (samdark) [inline composer.json]: Doesn't seem to be used.

File: composer.json [hunk 5]
```diff
@@ -18,20 +18,20 @@
...
+        "yiisoft/cache-file": "^3.0@dev",
```
Comment (samdark) [inline composer.json]: Doesn't seem to be used.

File: composer.json [hunk 6]
```diff
@@ -18,20 +18,20 @@
...
         "yiisoft/di": "^3.0@dev",
```
Comment (samdark) [inline composer.json]: Doesn't seem to be used.

File: composer.json [hunk 7]
```diff
@@ -18,20 +18,20 @@
...
         "yiisoft/log": "^3.0@dev",
```
Comment (samdark) [inline composer.json]: Doesn't seem to be used.

File: composer.json [hunk 8]
```diff
@@ -18,20 +18,20 @@
...
-        "yiisoft/yii-console": "^3.0@dev",
-        "yiisoft/yii-web": "^3.0@dev",
-        "hiqdev/composer-config-plugin": "^1.0@dev"
+        "yiisoft/yii-console": "^3.0@dev"
```
Comment (samdark) [inline composer.json]: Doesn't seem to be used.

**Comment by terabytesoftw** [comment]
Fixed tests:

- [x] ColumnSchemaBuilderTest.
- [x] CommandTest.
- [x] ConnectionTest.
- [x] QueryBuilderTest.
- [x] QueryTest.

File: .travis.yml [hunk 9]
```diff
@@ -29,13 +11,13 @@ language: php
...
     - DEFAULT_COMPOSER_FLAGS="--prefer-dist --no-interaction --no-progress --optimize-autoloader"
-    - TASK_TESTS_PHP=1
-    - TASK_TESTS_COVERAGE=0
     - TRAVIS_SECOND_USER=travis_two
 
 
 services:
+  - memcached
   - mysql
+  - postgresql
```
Comment (samdark) [inline .travis.yml]: Why do we need postgresql for MySQL package?

File: composer.json [hunk 10]
```diff
@@ -18,20 +18,14 @@
...
         "phpunit/phpunit": "^8.4",
-        "yiisoft/view": "^3.0@dev",
-        "yiisoft/cache": "^3.0@dev",
-        "yiisoft/active-record": "^3.0@dev",
-        "yiisoft/di": "^3.0@dev",
-        "yiisoft/log": "^3.0@dev",
-        "yiisoft/yii-console": "^3.0@dev",
-        "yiisoft/yii-web": "^3.0@dev",
-        "hiqdev/composer-config-plugin": "^1.0@dev"
+        "yiisoft/active-record": "^3.0@dev"
```
Comment (samdark) [inline composer.json]: Should not be required.

File: .travis.yml (hunk 9 above)
Comment (terabytesoftw) [reply .travis.yml]: Done.

File: composer.json (hunk 10 above)
Comment (terabytesoftw) [reply composer.json]: Done.

**Comment by terabytesoftw** [comment]
Fixed tests:

- [x] ActiveQueryTest.
- [x] BatchQueryResulTest.

**Comment by samdark** [comment]
Let's merge it and continue with smaller pull requests. Thanks, @terabytesoftw.

**Pull request merged**